lib/drives/pages/drive_detail_page.dart

```diff
@@ -353,14 +385,39 @@ class _DriveDetailPageState extends State<DriveDetailPage> {
     await _drive?.cancel();
     setState(() {});
   }
+
+  int _getNewMessagesCount(Drive drive) {
+    List<Ride> approvedRides = drive.approvedRides!;
+    int count = 0;
+    for (Ride ride in approvedRides) {
+      count += ride.getUnreadMessagesCount();
+    }
+    return count;
+  }
+
+  List<Widget> _pendingRidesList(List<Ride> pendingRides) {
+    List<Widget> pendingRidesColumn = [];
+    if (pendingRides.isNotEmpty) {
+      pendingRidesColumn = List.generate(
+        pendingRides.length,
+        (index) => PendingRideCard(
+          pendingRides.elementAt(index),
+          reloadPage: loadDrive,
+          drive: _drive!,
+        ),
+      );
+    }
+    return pendingRidesColumn;
+  }
 }
 
 class WaypointAction {
   final Profile profile;
   final bool isStart;
   final int seats;
+  final int rideId;
```

Comment:
Why was this needed? I don't see it used anywhere

Reply:
line 310

Reply:
Oh thanks! But now I'm confused why don't pass the whole ride? That would make both `seats` and `profile` unnecessary.

Reply:
?

Reply:
changed that